### PR TITLE
Added a toUInt32 method to the NativePointer

### PR DESCRIPTION
**Pull request by FrankSpierings**
If it turns out that I have done this correctly, I might do the UInt64 (and Int64) as well ;)

### CODE REVIEW

**Comment by oleavr** [comment]
Nice :)

Only thing before merging – could you fix the author so it's exactly like this: `Full Name <foo@bar.com>`. E.g. first squash then `git commit --amend --author="Full Name <foo@bar.com>`.

To avoid having to do this again, do something like:

```sh
$ git config --global user.name "Full Name"
$ git config --global user.email "foo@bar.com"
```

**Comment by oleavr** [comment]
Btw, this also needs a test. E.g. by copy-pasting [this one](https://github.com/frida/frida-gum/blob/27035d82554a993c3d159db4c6f2a1e1b83a0051/tests/gumjs/script.c#L1362-L1381) and adjusting. To run just that test, do `../build/tmp-macos-x86_64/frida-gum/tests/gum-tests -p /GumJS/Script/native_pointer_provides_arithmetic_operations#DUK`. (Replacing `DUK` with `V8` to test it on V8.)

**Comment by FrankSpierings** [comment]
I believe I have done what you have asked here as well (including running the tests). Please check my work.

**Comment by oleavr** [comment]
I actually need this functionality myself right now, so I'll go ahead and write the test for you so we can get this landed. 😊

**Comment by oleavr** [comment]
Landed as 5027e19bd5ff093c8926a2a2ef05a6ab4229545b. Thanks!

File: tests/gumjs/script.c [hunk 1]
```diff
@@ -1380,6 +1380,29 @@ SCRIPT_TESTCASE (native_pointer_provides_arithmetic_operations)
...
+      "send(ptr(1).sub(2).toUInt32());");
+  EXPECT_SEND_MESSAGE_WITH ("7");
+  EXPECT_SEND_MESSAGE_WITH ("3");
+  EXPECT_SEND_MESSAGE_WITH ("2");
+  EXPECT_SEND_MESSAGE_WITH ("7");
+  EXPECT_SEND_MESSAGE_WITH ("5");
+  EXPECT_SEND_MESSAGE_WITH ("3");
+  EXPECT_SEND_MESSAGE_WITH ("8");
+  EXPECT_SEND_MESSAGE_WITH ("0");
+  EXPECT_SEND_MESSAGE_WITH ("4294967295");
```
[the remark this answers is not in the document]
Comment (FrankSpierings) [reply tests/gumjs/script.c]: Thank you for the feedback. I will try to adhere to these next time ;)

**Pull request closed without merging**